packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/PagespeedInsights.astro

```diff
@@ -0,0 +1,160 @@
+
+runTestButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+
+    runTestButton.loading = true;
+    runTestButton.disabled = true;
+
+    console.log('PageSpeed Insights test started.', lighthouseFetchUrl)
+
+    const response = await fetch(lighthouseFetchUrl);
+
+    let performanceScore = 0;
+    let accessibilityScore = 0;
+    let bestPracticesScore = 0;
+    let seoScore = 0;
+
+    if (response.ok) {
+
+        const data = await response.json();
+
+        console.log('Successfully fetched data from the PageSpeed Insights API.', data);
+
+        performanceScore = data.lighthouseResult.categories.performance.score * 100;
+        accessibilityScore = data.lighthouseResult.categories.accessibility.score * 100;
+        bestPracticesScore = data.lighthouseResult.categories['best-practices'].score * 100;
+        seoScore = data.lighthouseResult.categories.seo.score * 100;
+
+        const perfRing = document.getElementById('perf-ring') as SlProgressRing
+        if (perfRing) {
+            perfRing.value = performanceScore;
+            perfRing.textContent = performanceScore.toString();
+            perfRing.style.color = performanceScore >= 90 ? 'green' : performanceScore >= 50 ? 'rgb(208, 208, 0)' : 'red';
+            perfRing.style.setProperty("--indicator-color", performanceScore >= 90 ? 'green' : performanceScore >= 50 ? 'yellow' : 'red');
+            if (performanceScore < 90) {
+                perfRing.style.webkitTextStroke = '2px black;'
+            }
+        }
+
+        const accRing = document.getElementById('acc-ring') as SlProgressRing
+        if (accRing) {
+            accRing.value = accessibilityScore;
+            accRing.textContent = accessibilityScore.toString();
+            accRing.style.color = accessibilityScore >= 90 ? 'green' : accessibilityScore >= 50 ? 'rgb(208, 208, 0)' : 'red';
+            accRing.style.setProperty("--indicator-color", accessibilityScore >= 90 ? 'green' : accessibilityScore >= 50 ? 'yellow' : 'red');
+            if (accessibilityScore < 90) {
+                accRing.style.webkitTextStroke = '2px black;'
+            }
+        }
+
+        const bpRing = document.getElementById('bp-ring') as SlProgressRing
+        if (bpRing) {
+            bpRing.value = bestPracticesScore;
+            bpRing.textContent = bestPracticesScore.toString();
+            bpRing.style.color = bestPracticesScore >= 90 ? 'green' : bestPracticesScore >= 50 ? 'rgb(208, 208, 0)' : 'red';
+            bpRing.style.setProperty("--indicator-color", bestPracticesScore >= 90 ? 'green' : bestPracticesScore >= 50 ? 'yellow' : 'red');
+            if (bestPracticesScore < 90) {
+                bpRing.style.webkitTextStroke = '2px black;'
+            }
+        }
+
+        const seoRing = document.getElementById('seo-ring') as SlProgressRing
+        if (seoRing) {
+            seoRing.value = seoScore;
+            seoRing.textContent = seoScore.toString();
```

Comment:
Can we make a helper function which assigns all this crap for us so we don't have to manually do it every time.

So a function which takes a ring element and does all this stuff

Reply:
[e7a2802](https://github.com/astrolicious/studiocms/pull/125/commits/e7a2802901b32720f59d3aeaf6fa22b06dd217d3) honestly don't know why, but I didn't expect that to actually work...

Reply:
moved in [a9192f9](https://github.com/astrolicious/studiocms/pull/125/commits/a9192f942255f31dbc708b9662a61e887239f153) to the util file